contracts/cw20-stake-external-rewards/src/contract.rs

```diff
@@ -463,6 +542,32 @@ pub fn execute_update_manager(
         ))
 }
 
+pub fn execute_update_pause_status(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    pause: bool,
+) -> Result<Response<Empty>, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+    if Some(info.sender.clone()) != config.owner && Some(info.sender) != config.manager {
+        return Err(ContractError::Unauthorized {});
+    };
```

Comment:
i really like how you've let both the manager and owner pause, but only the owner withdraw. seems like a good system where the manager can stop the world, but the owner can unstop it and withdraw.